lib/engine/step/train.rb

```diff
@@ -86,39 +93,51 @@ def available_cash(player)
       def buyable_trains(entity)
         depot_trains = @depot.depot_trains
         other_trains = @depot.other_trains(entity)
-        # If the corporation cannot buy a train, then it can only buy the cheapest available (if there is any...)
-        min_depot_train = @depot.min_depot_train
-        if min_depot_train && min_depot_train.price > entity.cash
 
-          depot_trains = [min_depot_train] if @game.class::EBUY_DEPOT_TRAIN_MUST_BE_CHEAPEST
+        if entity.cash < (@depot.min_depot_price || 0)
+          depot_trains = [@depot.min_depot_train] if @game.class::EBUY_DEPOT_TRAIN_MUST_BE_CHEAPEST
 
           if @last_share_sold_price
-            # 1889, a player cannot contribute to buy a train from another corporation
-            return depot_trains unless @game.class::EBUY_OTHER_VALUE
-
-            # 18Chesapeake and most others, it's legal to buy trains from other corps until
-            # if the player has just sold a share they can buy a train between cash-price_last_share_sold and cash
-            # e.g. If you had $40 cash, and if the train costs $100 and you've sold a share for $80,
-            # you now have $120 cash the $100 train should still be available to buy
-            min_cash_to_spend = (entity.cash + entity.owner.cash) - @last_share_sold_price
-            return depot_trains + (other_trains.reject { |t| t.price < min_cash_to_spend })
+            other_trains =
+              if @game.class::EBUY_OTHER_VALUE
+                must_spend = (entity.cash + entity.owner.cash) - @last_share_sold_price + 1
+                other_trains.reject { |t| t.price < must_spend }
+              else
+                []
+              end
           end
         end
-        depot_trains + other_trains
+
+        depot_trains.concat(other_trains)
```

Comment:
hm ya since you return it here, i'd probably keep it as a +, if you were doing more stuff after this with depot_trains, yea concat it